test/test_torch.py

```diff
@@ -2697,14 +2697,16 @@ def test_gradient_type_promotion(self, device):
                 self.assertEqual(actual, expected, equal_nan=True, exact_dtype=False)
 
     def _test_large_cum_fn_helper(self, x, fn):
-        x_cpu = x.cpu().float()
-        expected = fn(x_cpu)
+        expected = fn(x.cpu().float())
         actual = fn(x).cpu().float()
-        self.assertEqual(expected, actual.cpu().float())
+        # Direct comparison to save memory usage.
+        self.assertTrue((expected == actual).all())
```

Comment:
I don't think this change is equivalent.
`self.assertEqual` maps to `torch.testing.TestCase`'s assertEqual, and is overloaded to do relaxed-margin "equal" similar to `allclose`. It is defined here: https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/common_utils.py#L2813-L2896

assertTrue for `expected == actual` does exact float32 equality (and not based on relaxed margin equal)

Reply:
maybe you want to do `torch.allclose(expected, actual)` with the correct `atol` passed in?

Reply:
Sure I can do that. (will also need to update the memory usage as I found `allclose` consume an extra 4x memory of input.)

Reply:
Changed to `torch.testing.assert_allclose`. If I'm not mistaken this uses the same default tolerance that `self.assertEqual` uses.